### ISSUE TITLE
Size mismatch exception in model's forward when using larger input sizes

### ISSUE THREAD

**Issue by mirceast** [NullConvergence/torch-template#30]
**Description:**

When increasing the input size to more than (60, 60), a size mismatch exception occurs. The below exception doesn't occur when the input size is (50,  50).

**Exact steps to reproduce:**

- Just to avoid OOM issues, change "ResNet34" to "ResNet18" in  `configs/runs/cifar10.json`
- In `torch_temp/data/data_loaders/vision/cifar10.py`
change `transforms.RandomCrop(32, padding=4)` (line 14) to 
`transforms.Resize((100, 100))`
- The error should appear before any validation starts, but one can add 
`transforms.Resize((100, 100))` in line 23 just to be thorough.
- `python train.py --config configs/runs/cifar10.json`

**Output:**
```
Exception has occurred: RuntimeError
size mismatch, m1: [128 x 4608], m2: [512 x 10] at /opt/conda/conda-bld/pytorch_1565272271120/work/aten/src/THC/generic/THCTensorMathBlas.cu:273
  File "/home/mircea/.conda/envs/logo/lib/python3.7/site-packages/torch/nn/functional.py", line 1369, in linear
    ret = torch.addmm(bias, input, weight.t())
  File "/home/mircea/.conda/envs/logo/lib/python3.7/site-packages/torch/nn/modules/linear.py", line 87, in forward
    return F.linear(input, self.weight, self.bias)
  File "/home/mircea/.conda/envs/logo/lib/python3.7/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mircea/HKT/torch_temp/torch_temp/model/blocks/resnet.py", line 97, in forward
    out = self.linear(out)
  File "/home/mircea/.conda/envs/logo/lib/python3.7/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mircea/HKT/torch_temp/torch_temp/trainer/default.py", line 101, in _run_batch
    output = self.model(data)
  File "/home/mircea/HKT/torch_temp/torch_temp/trainer/default.py", line 65, in _train_epoch
    loss = self._run_batch(data, target)
  File "/home/mircea/HKT/torch_temp/torch_temp/trainer/base.py", line 50, in train
    result = self._train_epoch(epoch)
  File "/home/mircea/HKT/torch_temp/train.py", line 100, in main_normal
    trainer.train()
  File "/home/mircea/HKT/torch_temp/train.py", line 130, in <module>
    main_normal()
  File "/home/mircea/.conda/envs/logo/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/mircea/.conda/envs/logo/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/home/mircea/.conda/envs/logo/lib/python3.7/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
```

**Comment by xserban** [comment]
Thanks for reporting!
I solved the issue in: https://github.com/NullConvergence/torch_temp/pull/34 and merge in develop.
Please check the result and reopen the issue if you have any problems.
